utils/utils.js

```diff
@@ -171,7 +167,56 @@ const volunteerForNewGiftees = async (user, count) => {
   await batch.commit()
 
   // return the result after the database stuff is complete
-  return { success: result }
+  return { success: "Please refresh the giftee list" }
+}
+
+/**
+ * @param {object} giftee - details about the giftee
+ * @param {string} [giftee.uuid] - UUID of the giftee, if you do not know it
+ * @param {string} [giftee.user] - If the UUID is unknown this is the giftee's uid or user object
+ * @param {object} update - details about the giftee
+ * @param {string} update.field - What part to mark: sent, received, reported
+ * @param {string} [update.message] - If reporting allow a message
+ * @param {boolean} [update.value] - If reporting allow a message
+ * @returns
+ */
+async function markGifteeAccount({uuid, user}, {field, message, value}){
+  // if someone is marking the gift sent they know the uuid of the giftee
+  if(!uuid){
+    if(!user){return {error: "no uuid or user requested"}}
+
+    // if the uuid is undefined then take the user, search for teh account related and return that uuid
+    let tmp_uuid = await getUUID(user)
+    if(tmp_uuid.error){return {error: "no API key set"}}
+    uuid = tmp_uuid.success
+  }
+
+  // checking teh field
+  if(!field){return {error: "no field defined"}}
+  if(field !== "sent" && field !== "received" && field !== "reported"){return {error: "field is not one of the defined types"}}
+
+  if(typeof value === "undefined"){value=true}
+  let currentValueRaw = await db.collection('events').doc(EVENT).collection('participants').doc(uuid).get()
+  if (!currentValueRaw.exists) {return {error: "no such user"}}
+  let currentValue = currentValueRaw.data()
+  if(value === currentValue[field]){return {success: "Value already set"}}
+
+  let tmp = {}
+  tmp[field] = value
+  if(message){tmp.report = message}
+
+  let entryResult = await db.collection('events').doc(EVENT).collection('participants').doc(uuid).set(tmp, {merge: true}).then(()=> {return true}).catch(() => {return false});
```

Comment:
But is it the _account_ that is to be marked? Isn't it the _gift_ related to the account?

We have to support multiple gifts, so the status of the gift must be set on the gift.

Reply:
Current setup is that its on teh participation entry (https://github.com/nightzirch/secret-toymaker.backend/blob/develop_silver/https/participate.js) that its marked, so an account can recieve one gift but send multiple

Reply:
And yes I know that is nowhere near perfect, I just want a working system for the current time

Reply:
I do really need to rethink how the layout fill work for future stuff, a nice refactor will be in order

Reply:
That can be a future upgrade.

Is the `gifter` prop the participant's secret toymaker? We gotta make sure to hide that :D I'll see if we can set up some rules in Firestore to never reveal those data.

Reply:
The file I linked? Ye the gifter is who is their gifter, iff the top of my head I dont think it is retuened to teh user at all, I think that entire object is used for backend only, Ill check tomorrow